log.js

```diff
@@ -13,57 +13,36 @@ winston.addColors({
   debug: 'grey',
 })
 
-logger.clear()
-if (process.env.NODE_ENV === 'test') {
-  logger.add(winston.transports.File, {
-    name: 'log',
-    level: 'debug',
-    filename: 'tests.log',
-    json: false,
-    colorize: true,
-    prettyPrint: true,
-  })
-} else if (process.env.NODE_ENV === 'production') {
-  logger.add(winston.transports.Console, {
-    name: 'log',
-    level: 'info',
-    json: true,
-    colorize: false,
-    prettyPrint: false,
-    silent: false,
-    timestamp: true,
-    stringify: true,
-    handleExceptions: true,
+const flattened = winston.format((meta) => {
+  const { message, level, timestamp: ts, ...rest } = meta
+  const result = flattenMeta(rest)
+  Object.entries(result).forEach(([key, val]) => {
+    // eslint-disable-next-line no-param-reassign
+    meta[key] = val
   })
-} else {
-  logger.add(winston.transports.Console, {
-    name: 'log',
-    level: 'info',
-    json: false,
-    colorize: true,
-    // prettyPrint: true,
-    silent: false,
-    timestamp: true,
-    handleExceptions: true,
-    humanReadableUnhandledException: true,
-  })
-}
+  return meta
+})
+
+const format =
+  process.env.NODE_ENV === 'test'
+    ? combine(timestamp(), colorize(), simple(), prettyPrint())
+    : combine(timestamp(), flattened(), json())
+
+const logger = winston.createLogger({
+  format,
+  transports: [new winston.transports.Console({ level: 'info', handleExceptions: true })],
+})
 
 if (appInsights) {
   logger.info('Activating application insights logger')
 
   logger.add(
```

Comment:
Definitely need to be careful around testing logging with this change as the api changed significantly and support for application insights seems a bit precarious.